concepts/strings/about.md

```diff
@@ -0,0 +1,265 @@
+# Introduction
+
+Characters and Strings form the basis for text representation in Elm.
+Strings can be thought of as lists of characters and most `List` functions have their equivalent in the `String` module.
+
+## Characters
+
+The `Char` type represents a single unicode character with single quotes e.g. 'a'. 
+The `Char` module provides predicate functions (`Char -> Bool`) to test qualities of a character: `isUpper`, `isLower`, `isAlpha`, and `isAlphaNum`
+The module also provides functions to convert a character `toUpper`, `toLower`, `toLocaleUpper`, `toLocaleLower`, as well as to/from their numerical unicode value (`Int`) with `toCode` and `fromCode`.
+
+## Strings
+
+The `String` type provides a built-in representation for efficient string manipulation and can represent any sequence of unicode characters.
+Strings are represented with double quotes: `"this is a string"`.
+Multi-line strings are represented with triple double quotes and can have unescaped quotes and newlines.
+
+```elm 
+haiku = 
+"""
+    Code's strings entwine
+    , In binary, thoughts combine
+    , Logic's elegant line.
+"""
+```
+
+## String Manipulation
+
+`reverse : String -> String` reverses the string.
+```elm 
+reverse "Hello world." 
+    --> ".dlrow olleH"
+```
+
+`repeat : Int -> String -> String` repeats a string n times.
+```elm 
+repeat 2 "Hello world." 
+    --> "Hello world.Hello world."
+```
+
+`replace : String -> String -> String -> String` replaces all occurrences of a given substring.
+```elm
+replace "Hello" "Goodbye cruel" "Hello world." 
+    --> "Goodbye cruel world."
+```
+
+`append : String -> String -> String` appends two strings.
+The `++` operator can also be used to do this.
+```elm
+append "butter" "fly" 
+    --> "butterfly"
+"butter" ++ "fly" 
+    --> "butterfly"
+```
+
+`concat : List String -> String` concatenates a list of strings into a single string.
+```elm
+concat ["never","the","less"] 
```

Comment:
I think it would be good to format all the arrays consistently.

Mostly there are no spaces (`["1","2"]`) in this document, but occasionally there are spaces (`["1", "2"]`).

I think it probably also makes sense to go with whatever elm-format does, which I think is `[ "1", "2" ]`.

Other than this I think the concept looks great.

Reply:
Thanks for the feedback Cedd.

I updated the lists' format and the other recommendations. I think this is ready to merge.

What do you think of using the "99 Bottles of Beer" problem for this concept? I feel it is simple enough that the focus will be in the use of String. My only concern is that it requires recursion which is not included in the concepts yet, although it is used in other exercises.

Reply:
Hi Pete, until now we have always created new exercises when we have made concepts, so I don't know what the repercussions would be of using an existing one. I think you would have to duplicate it as a concept exercise, which might be a bit strange. Maybe we could deprecate the old one or something.

Anyway, hopefully there is a better way. This page shows a list of concepts and their associated concept exercises across all the exercism tracks.
https://exercism.org/docs/building/tracks/stories

Quite a few tracks have already implemented a string concept, so it is probably easier to repurpose one / copy paste one. And that will hopefully make it easier to work in more of the string functions.

I'm going to wait for Matthieu or Jie to approve this before merging, which sounds like it might take a week, apologies for that. I don't think there will be many changes after their review though, so you could start on the concept exercise now if you want, probably in a new branch.

Reply:
Cedd, no problem waiting for the review. 

IRT using 99-Beers, I do not see it in the current Elm exercises. Is it by another name? I was not able to find anything similar. Maybe I worded the question poorly, what I meant was whether you thought that problem was well suited, not that it was already written.

I started working on 99 Beers and it seems a good fit to introduce the basics, but more exercises will be needed to touch more of the String module functionality. I will open a different issue and PR for that.

Reply:
Ah I see. I've done that 99 beers problem in other languages so just assumed it was implemented in Elm, but obviously  not :)